Add ability to specify timeout and max content length for HTTP server

whummer opened the pull request:
Add ability to specify request timeout and max content length for HTTP server. Preparation for a fix for #5598

thrau reviewed (approved):
looks useful! the name `timeout` is a bit ambiguous in a server context maybe.

whummer commented:
Good point. Renamed to `send_timeout`, added a docstring with some details about the parameters, and a simple integration test for the max content length..
